src/Param.cc

```diff
@@ -828,27 +977,47 @@ bool Param::SetParentElement(ElementPtr _parentElement)
 void Param::Reset()
 {
   this->dataPtr->value = this->dataPtr->defaultValue;
-  this->dataPtr->strValue = this->dataPtr->defaultStrValue;
+  std::string strValue;
+  SDF_ASSERT(this->dataPtr->StringFromValueImpl(PrintConfig(),
```

Comment:
You are right, that would be the ideal behavior. It was this way in the past to prevent uninitialized values. Thanks for catching it, fixed in https://github.com/ignitionrobotics/sdformat/pull/729/commits/e903b131aa582f33931d32094e140bd1836a5890